### PR TITLE
[ILM] HLRC-ILM  Retry Lifecycle Policy docs

### PR DESCRIPTION
**Pull request by talevy**
adds HLRC documentation for the ILM retry lifecycle policy action

this PR also renames `retryLifecycleStep` to `retryLifecyclePolicy` in the index-lifecycle-client

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java [hunk 1]
```diff
@@ -201,8 +206,7 @@ public void testGetLifecyclePolicy() throws IOException, InterruptedException {
         ActionListener<GetLifecyclePolicyResponse> listener =
             new ActionListener<GetLifecyclePolicyResponse>() {
                 @Override
-                public void onResponse(GetLifecyclePolicyResponse response)
-                {
+                public void onResponse(GetLifecyclePolicyResponse response) {
```
Comment (gwbrown) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java]: I'm a little surprised this passes the length check, I thought it was over. Cool if it is though!

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java [hunk 2]
```diff
@@ -226,6 +230,82 @@ public void onFailure(Exception e) {
...
+        // setup policy to immediately fail on index
+        {
+            Map<String, Phase> phases = new HashMap<>();
+            Map<String, LifecycleAction> warmActions = new HashMap<>();
+            warmActions.put(ShrinkAction.NAME, new ShrinkAction(1));
+            phases.put("warm", new Phase("warm", TimeValue.ZERO, warmActions));
+
+            Map<String, LifecycleAction> deleteActions =
+                Collections.singletonMap(DeleteAction.NAME, new DeleteAction());
+            phases.put("delete", new Phase("delete",
```
Comment (gwbrown) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java]: What's the reasoning behind having a delete phase here?

File: docs/java-rest/high-level/ilm/retry_lifecycle_policy.asciidoc [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+==== Request
+
+The Retry Lifecycle Policy API allows you to invoke execution of policies
+that encountered errors in certain indices.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request]
+--------------------------------------------------
+<1> retries execution of `my_index`'s policy
```
Comment (gwbrown) [inline docs/java-rest/high-level/ilm/retry_lifecycle_policy.asciidoc]:
```suggestion
<1> Retries execution of `my_index`'s policy
```

**Review by AthenaEryma — APPROVED** [review]
Left a couple nits, otherwise LGTM

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java (hunk 1 above)
Comment (talevy) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java]: oh, that is my bad. probably happened on accident via rebase

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java (hunk 2 above)
Comment (talevy) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java]: true, no need. didn't think it hurts. but it is also unnecessary LOC

File: docs/java-rest/high-level/ilm/retry_lifecycle_policy.asciidoc (hunk 3 above)
Comment (talevy) [reply docs/java-rest/high-level/ilm/retry_lifecycle_policy.asciidoc]: 👍

**Pull request merged**